lib/stack_master/parameter_resolvers/latest_ami.rb

```diff
@@ -0,0 +1,25 @@
+module StackMaster
+  module ParameterResolvers
+    class LatestAmi < Resolver
+      array_resolver class_name: 'LatestAmis'
+
+      def initialize(config, stack_definition)
+        @config = config
+        @stack_definition = stack_definition
+        @ami_finder = AmiFinder.new(@stack_definition.region)
+      end
+
+      def resolve(value)
+        case value
+        when String
+          owners = ['self']
+          filters = @ami_finder.build_filters_from_string(value)
+        when Hash
+          owners = Array(value.delete('owner_id').to_s || 'self')
+          filters = @ami_finder.build_filters_from_hash(value)
+        end
```

Comment:
Why support both string and hash options?

I'm not sure what value the hash options gives in this implementation, because the hash and the key value string are essentially the same thing, the string just needs parsing into a hash. I thought we wanted to use a hash to be able to nest a level deeper than the key value string would allow.